### PR TITLE
[WIP]  Create SequenceClassification MultipleChoice and TokenClassification for tf_longformer

### LINKED ISSUE #6401: [TF Longformer] Add Multiple Choice, Seq Classification Model
# 🚀 Feature request

`modeling_longformer.py` has the classes `LongformerForSequenceClassification`, `LongformerForMultipleChoice` and `LongformerForTokenClassification` which are not present  in `modeling_tf_longformer.py` at the moment.
Those classes should be equally added to `modeling_tf_longformer.py`.
 
## Motivation

The pretrained weights for TFLongformer are available so that these classes could be used for finetuning.

## Your contribution

This issue is a good first issue because it is not too complicated to add these models. One should take a look at `modeling_tf_roberta.py` to see how these models are implemented for `TFRoberta` and implement them analogous for `TFLongformer`. Please make sure that the docstring is correct and that test are added for each class (again Roberta can serve as an example here, check out `test_modeling_tf_roberta.py`).

I am happy to guide interested community contributors through the PR and help them get it merged.

### PR DESCRIPTION
**Pull request by Groskilled**
Tests for the new classes are not passing and there is no test yet for TFLongformerClassificationHead (and I am not sure it's needed).
@patrickvonplaten 

Resolves #6401

### CODE REVIEW

File: src/transformers/modeling_tf_longformer.py [hunk 1]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+            activation="tanh",
+            name="dense",
+        )
+        self.dropout = tf.keras.layers.Dropout(config.hidden_dropout_prob)
+        self.out_proj = tf.keras.layers.Dense(
+            config.num_labels, kernel_initializer=get_initializer(config.initializer_range), name="out_proj"
+        )
+
+    def call(self, features, training=False):
+        x = features[:, 0, :]
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_longformer.py:1304]: Can you please rename both `fetures` and `x` to `hidden_states`. We try to avoid single letter variable names.

File: src/transformers/modeling_tf_longformer.py [hunk 2]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+            1, kernel_initializer=get_initializer(config.initializer_range), name="classifier"
+        )
+
+    @property
+    def dummy_inputs(self):
+        """ Dummy inputs to build the network.
+        Returns:
+            tf.Tensor with dummy inputs
+        """
+        return dict(input_ids=tf.constant(MULTIPLE_CHOICE_DUMMY_INPUTS))
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_longformer.py:1493]: This variable is not defined at the moment `MULTIPLE_CHOICE_DUMMY_INPUTS` -> it should be similar to the one of other models, but also include a `global_attention_mask` variable similar to the dummy inputs of `TFLongformerPreTrainedModel`

File: src/transformers/modeling_tf_longformer.py [hunk 3]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+            if len(inputs) > 9:
+                inputs = inputs[:9]
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            labels = inputs.pop("labels", labels)
+
+        if global_attention_mask is None:
+            logger.info("Initializing global attention on CLS token...")
+            global_attention_mask = tf.zeros_like(inputs)
+            # global attention on cls token
+            global_attention_mask[:, 0] = 1
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_longformer.py:1366]: this does not work in tf -> you will have to work with `tf.where()` here or leave it and I'll take a look at the end.

File: src/transformers/modeling_tf_longformer.py [hunk 4]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+    ):
+        r"""
+        labels (:obj:`tf.Tensor` of shape :obj:`(batch_size, sequence_length)`, `optional`, defaults to :obj:`None`):
+            Labels for computing the token classification loss.
+            Indices should be in ``[0, ..., config.num_labels - 1]``.
+        """
+        if not return_dict:
+            return_dict = self.longformer.return_dict
+
+        if global_attention_mask is None:
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_longformer.py:1439]: same we need to check if `global_attention_mask` is in the dict or not. You can check out TFLongformerForQuestionsAnswering` to see how it is done there.

File: src/transformers/modeling_tf_longformer.py [hunk 5]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+            logger.info("Initializing global attention on CLS token...")
+            global_attention_mask = tf.zeros_like(inputs)
+            # global attention on cls token
+            global_attention_mask[:, 0] = 1
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_longformer.py:1443]: does not work in TF

**Comment by patrickvonplaten** [comment]
Hey @Groskilled - I added a couple of comments. Let's try to first make the torch tests pass. Let me know if you need more help after implementing the comments.

File: src/transformers/modeling_tf_longformer.py (hunk 3 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_tf_longformer.py:1366]: This won't be possible tf -> np -> tf conversion is quite expensive and very slow on GPU. You will have to use something like `tf.where((tf.range(sequence_length) < 1).broatcast_to(shape_list(global_attention_mask), 1, global_attention_mask)`. You could take a look into other `tf.where` statements in TF Longformer to understand better.

File: src/transformers/modeling_tf_longformer.py (hunk 4 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_tf_longformer.py:1439]: `inputs` can be a dict. In this case `global_attention_mask` could be a key in the dict, but would be set to None from the default keyword argument -> thus one should check if `global_attention_mask` is in the dict or not and if yes set it to its value

File: src/transformers/modeling_tf_longformer.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_tf_longformer.py:1493]: I think this looks good -> you can check if the tests pass or not

File: src/transformers/modeling_tf_longformer.py [hunk 6]
```diff
@@ -1274,6 +1284,305 @@ def call(
...
+            return_dict = self.longformer.return_dict
+
+        if isinstance(inputs, (tuple, list)):
+            labels = inputs[9] if len(inputs) > 9 else labels
+            if len(inputs) > 9:
+                inputs = inputs[:9]
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            labels = inputs.pop("labels", labels)
+
+        if global_attention_mask is None:
```
[the remark this answers is not in the document]
Comment (Groskilled) [reply src/transformers/modeling_tf_longformer.py:1362]:
```suggestion
        elif isinstance(inputs, (dict, BatchEncoding)):
            labels = inputs.pop("labels", labels)
            global_attention_mask = inputs.get("global_attention_mask", None)
            labels.pop("labels", labels)
            
        if global_attention_mask is None:
```

File: src/transformers/modeling_tf_longformer.py (hunk 3 above)
Comment (Groskilled) [reply src/transformers/modeling_tf_longformer.py:1366]: Is the attention mask always 2 dim or could it be anything ? I wrote something but it's not working with everything.

**Comment by patrickvonplaten** [comment]
Closing PR due to inactivity
